[Core] Load tries to load non-packages

# Command bugs

#### Command name

[p]load

#### What cog is this command from?

Core

#### What were you expecting to happen?

For the bot to say that the module wasn't found in any cog path.

#### What actually happened?

```
[2020-11-18 13:02:41] [ERROR] red: Package loading failed
Traceback (most recent call last):
  File "c:\envs\joker3.4\lib\site-packages\redbot\core\core_commands.py", line 172, in _load
    await bot.load_extension(spec)
  File "c:\envs\joker3.4\lib\site-packages\redbot\core\bot.py", line 1202, in load_extension
    lib = spec.loader.load_module()
TypeError: load_module() missing 1 required positional argument: 'fullname'
```

#### How can we reproduce this issue?

Use `[p]load locales`

--- a/redbot/core/_cog_manager.py
+++ b/redbot/core/_cog_manager.py
@@ -258,6 +258,12 @@ class CogManager:
 
         try:
             mod = import_module(real_name, package=package)
+            if mod.__spec__.name == "redbot.cogs.locales":
+                raise NoSuchCog(
+                    "No core cog by the name of '{}' could be found.".format(name),
+                    path=mod.__spec__.origin,
+                    name=name,
+                )
         except ImportError as e:
             if e.name == package + real_name:
                 raise NoSuchCog(

